ipv8/test/REST/rtest/test_rest_api_peer.py

```diff
@@ -0,0 +1,260 @@
+import os
+import threading
+import time
+
+import logging
+
+from ipv8.REST.rest_manager import RESTRequest
+from ipv8.REST.root_endpoint import RootEndpoint
+from ipv8.configuration import get_default_configuration
+from ipv8.taskmanager import TaskManager
+from ipv8.test.REST.rtest.peer_communication import GetStyleRequests, PostStyleRequests
+from ipv8.test.REST.rtest.rest_peer_communication import HTTPGetRequester, HTTPPostRequester
+from ipv8_service import IPv8
+
+from twisted.web import server
+from twisted.internet.defer import maybeDeferred, inlineCallbacks, returnValue
+from twisted.internet import reactor
+
+
+class TestPeer(object):
+
+    def __init__(self,
+                 path,
+                 port,
+                 interface='127.0.0.1',
+                 configuration=None):
+
+        self._logger = logging.getLogger(self.__class__.__name__)
+        self._logger.info("Peer starting-up.")
+
+        self._rest_manager = None
+
+        self._port = port
+        self._interface = interface
+
+        self._path = path
+        self._configuration = configuration
+
+        # Check to see if we've received a custom configuration
+        if configuration is None:
+            # Create a default configuration
+            self._configuration = get_default_configuration()
+
+            self._configuration['logger'] = {'level': "ERROR"}
+
+            overlays = ['AttestationCommunity', 'IdentityCommunity']
+            self._configuration['overlays'] = [o for o in self._configuration['overlays'] if o['class'] in overlays]
+            for o in self._configuration['overlays']:
+                o['walkers'] = [{
+                    'strategy': "RandomWalk",
+                    'peers': 20,
+                    'init': {
+                        'timeout': 60.0
+                    }
+                }]
+
+        self._create_working_directory(self._path)
+        self._logger.info("Created working directory.")
+        os.chdir(self._path)
+
```

Comment:
Please overwrite the AttestationCommunity and IdentityCommunity `master_peer` so you do not interfere with the live network (this testsuite was firing requests to the live server :) ).